### PR TITLE
Remove ptmalloc2 hooks from v2.1.x

### PR DESCRIPTION
**Pull request by hjelmn**
In v2.0.x we added new memory hooks to replace ptmalloc2. We let the old hooks stick around but disabled them by default. We agreed to remove the hooks entirely in v2.1.x. If that is still the case please merge this PR.

### CODE REVIEW

File: opal/mca/btl/openib/btl_openib_component.c [hunk 1]
```diff
@@ -2914,6 +2881,22 @@ btl_openib_component_init(int *num_btl_modules,
         mca_btl_openib_component.if_exclude_list = NULL;
     }
 
+    /* If we are using ptmalloc2 and there are no posix threads
+       available, this will cause memory corruption.  Refuse to run.
+       Right now, ptmalloc2 is the only memory manager that we have on
+       OS's that support OpenFabrics that provide both FREE and MUNMAP
+       support, so the following test is [currently] good enough... */
```
Comment (jsquyres) [inline opal/mca/btl/openib/btl_openib_component.c]: @hjelmn You just deleted the ptmalloc component, so is this still needed?

**Comment by jsquyres** [comment]
@hppritcha and I discussed: this is a code cleanup PR, and is ok for v2.1.0.

**Comment by jsquyres** [comment]
bot:lanl:retest

**Comment by jsquyres** [comment]
@gpaulsen Can IBM test / review this?

**Comment by hppritcha** [comment]
bot:lanl:retest

**Comment by gpaulsen** [comment]
Well, I read the code, but didn't fully understand it.  I'll download it and give it a whirl.

**Comment by gpaulsen** [comment]
Sorry, getting back to this.  I'll do this today.

File: opal/mca/btl/openib/btl_openib_component.c (hunk 1 above)
Comment (gpaulsen) [inline opal/mca/btl/openib/btl_openib_component.c]: And it may not be true anymore.

**Comment by jsquyres** [comment]
Correction: review addressed in #2804.

**Pull request merged**